sw/device/silicon_creator/manuf/skus/earlgrey_a0/generic/ft_personalize_3.c

```diff
@@ -0,0 +1,84 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/crypto/drivers/entropy.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/json/provisioning_data.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/lib/testing/test_framework/ottf_test_config.h"
+#include "sw/device/lib/testing/test_framework/ujson_ottf.h"
+#include "sw/device/silicon_creator/lib/attestation.h"
+#include "sw/device/silicon_creator/lib/attestation_key_diversifiers.h"
+#include "sw/device/silicon_creator/lib/drivers/keymgr.h"
+#include "sw/device/silicon_creator/lib/error.h"
+#include "sw/device/silicon_creator/lib/otbn_boot_services.h"
+
+OTTF_DEFINE_TEST_CONFIG(.enable_uart_flow_control = true);
+
+/**
+ * Crank the keymgr to produce the attestation keys and certificates.
+ */
+static status_t personalize(ujson_t *uj) {
+  // Advance keymgr to Initialized state.
+  TRY(entropy_complex_init());
+  keymgr_advance_state();
+  TRY(keymgr_state_check(kKeymgrStateInit));
+
+  // Load OTBN attestation keygen program.
+  attestation_public_key_t curr_pubkey = {.x = {0}, .y = {0}};
+  TRY(otbn_boot_app_load());
+
+  // Advance keymgr and generate UDS attestation keys / cert.
+  // TODO(#19455): set attestation binding to OTP *Cfg partition measurements.
+  keymgr_advance_state();
+  TRY(keymgr_state_check(kKeymgrStateCreatorRootKey));
+  TRY(otbn_boot_attestation_keygen(kUdsAttestationKeySeed,
+                                   kUdsKeymgrDiversifier, &curr_pubkey));
+  // TODO(#19455): create certificate but do not sign it, it will be endored
```

Comment:
Is there value in initializing the UDS certificate as self-signed to verify the signature on the host side? Otherwise, we should add a comment describing how we are verifying the integrity of the key. 

A separate comment should mention that this process is expected to take place in a secure environment, as we are currently not doing any further authentication of the device before generating the endorsement certificate.

Reply:
Ah yes, good point, we can self-sign for integrity.

Added comment r.e. secure environment.